### ISSUE TITLE
MD044: URLs in HTML comments are flagged from 'names' configuration, even if wrapped in angle brackets

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#457]
### rumdl version

0.1.26

### Installation method

mise

### Bug description

Content in HTML comments can be checked by MD044 (https://github.com/rvben/rumdl/blob/2d34df06e8361b822708099a38e3f5940fd81e85/docs/md044.md). This is good because it often contains prose we might want to keep to a standard.

However, unlike regular Markdown, angle brackets can't be used to escape the check for URLs. Ideally, what is possible in regular Markdown should be possible in HTML comments.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Level 1 heading
   ---

   <https://www.example.test>

   <!-- This is a Test https://www.example.test -->
   <!-- This is a Test <https://www.example.test> -->
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [MD044]
   names = ["Test"]
   ```
1. Run `rumdl check` on the Markdown file.

### Expected behavior

Given the Markdown above, I'd expect violation to be flagged on line 7, where it's either:

- Treated as plain text and subject to MD044. This is current state.
- Treated as a bare URL and subject to MD034 (https://github.com/rvben/rumdl/blob/2d34df06e8361b822708099a38e3f5940fd81e85/docs/md034.md). This might be preferred state and would make the content inside HTML comments more like regular content.

I wouldn't expect a violation on line 8 because the `<` and `>` should escape any prose checks (and pass MD034).

### Actual behavior

An MD044 violation is found on line 7 and line 8.

### Configuration

```toml
[MD044]
names = ["Test"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.27 @eread. Angle-bracket autolinks in HTML comments are now detected using CommonMark's autolink pattern and excluded from proper name checking.

**Comment by eread** [comment]
> Fixed in v0.1.27 [@eread](https://github.com/eread). Angle-bracket autolinks in HTML comments are now detected using CommonMark's autolink pattern and excluded from proper name checking.

@rvben Yep, works well!
